packages/relay/src/lib/eth.ts

```diff
@@ -571,7 +571,34 @@ export class EthImpl implements Eth {
 
               // The block is from the last 15 minutes, therefore the historical balance hasn't been imported in the Mirror Node yet
               if (timeDiff < constants.BALANCES_UPDATE_INTERVAL) {
-                throw predefined.UNKNOWN_HISTORICAL_BALANCE;
+
+                const balance = await this.mirrorNodeClient.getBalanceAtTimestamp(mirrorAccount.account, undefined, requestId);
+                let currentBalance = 0;
+                let currentTimestamp;
+                let balanceFromTxs = 0;
+
+                if (balance.balances?.length) {
+                  currentBalance = balance.balances[0].balance;
+                  currentTimestamp = balance.timestamp;
+                }
+
+                let transactionsInTimeWindow = await this.mirrorNodeClient.getTransactionsForAccount(
+                    mirrorAccount.account,
+                    block.timestamp.to,
+                    currentTimestamp,
+                    requestId
+                );
+
+                for(const tx of transactionsInTimeWindow) {
+                  for (const transfer of tx.transfers) {
+                    if (transfer.account === mirrorAccount.account && !transfer.is_approval) {
+                      balanceFromTxs += transfer.amount;
+                    }
+                  }
+                }
+
+                balanceFound = true;
+                weibars = (currentBalance - balanceFromTxs) * constants.TINYBAR_TO_WEIBAR_COEF;
```

Comment:
Assuming a balance file timestamp of T1 and a block timestamp of T2 and these are within 15 mins of each other.
Can we make sure we have test coverage for the following cases

- T1 < T2
- T1 > T2
- T1 = T2

Also the above where
- it's a mixture of debits and credits
- it's all debits
- it's all credits

Reply:
Can do these in `eth.spec.ts` no need for it in acceptance test